source/core/TDCodecVia.cpp

```diff
@@ -936,8 +936,11 @@ Int32 TDViaParser::ParseArrayData(TypedArrayCoreRef pArray, void* pFirstEltInSli
                     // For nested dimensions just parse the next inner dimension using the array type.
                     subErr = ParseArrayData(pArray, pElement, level + 1);
                 }
-                if (subErr && !errCode)
+                if (subErr && !errCode) {
                     errCode = subErr;
+                    if (Fmt().StopArrayParseOnFirstError())
```

Comment:
I think we need to break out in general and not just for the JSON scenario. My concern is that the hang may still be there with bad VIA array code and we should fix that case too. I'm fine with this fix giving that we don't have much time to exercise it. Jared, please open an issue in Github to revisit this fix later on.